### PR TITLE
SOLR-15891: Remove bloat in Solr docker layers

### PR DESCRIPTION
**Pull request by HoustonPutman**
https://issues.apache.org/jira/browse/SOLR-15891

This removes the bloat in the first non-header docker layer, specifically this line:
```
find /opt/solr/ -type f -print0 | xargs -0 chmod 0644; \
```
That line increased the size of that docker layer to the complete size of the solr installation, meaning that two layers (in both local and official) were the full size of the solr installation.

By moving the 0644 chmod to the COPY and download sections, in local and official, the only thing we have to do in the referenced layer is give executables and directories the correct permissions, which in practice takes `250kB` instead of `200mB`. This achieves what we want.

One caveat is that `COPY --chmod` requires buildkit enabled, and Docker v20 installed. I think this is fine given this is a new offering, so we can assume users are using new Docker versions. (much better than having the docker image be 1.5-2x the size)

Some other small changes made:
- The tests now re-run when the `shared.sh` file is changed
- The test official image is tagged when it is built. We can change this in the future if we want to. (Maybe we re-think this in the future, and have a flag for the docker build to be "official" or "local", and keep the same commands (disabling push for "official".))
- I fixed the labels for the docker image
- The test directories that use `setfacl` on linux, now use `chmod a+rwx` if `setfacl` is not available. This is unfortunate, but there really is no other work around if we want the tests to run on Macs. Willing to have discussion here if y'all don't think this is acceptable.

### CODE REVIEW

File: solr/docker/templates/Dockerfile.official.header.template [hunk 1]
```diff
@@ -90,5 +90,6 @@ RUN set -ex; \
   fi; \
   { command -v gpgconf; gpgconf --kill all || :; }; \
   rm -r "$GNUPGHOME"; \
-  tar -C /opt --extract --file "/opt/solr-$SOLR_VERSION.tgz";
+  tar -C /opt --extract --preserve-permissions --file "/opt/solr-$SOLR_VERSION.tgz"; \
+  find "/opt/solr-$SOLR_VERSION/" -type f -print0 | xargs -0 chmod 0644;
```
Comment (janhoy) [inline solr/docker/templates/Dockerfile.official.header.template:95]: Speaking of saving space, could we uninstall gpg at the end of this `RUN` command? I.e. `apt-get --purge remove gpg && rm -rf /var/lib/apt/lists/*;` at the end?

**Review by janhoy — COMMENTED** [review]
Have not tested it, just some initial comments

File: solr/docker/templates/Dockerfile.official.header.template (hunk 1 above)
Comment (HoustonPutman) [reply solr/docker/templates/Dockerfile.official.header.template:95]: Just tried this out and it didn't affect this size of the layer, so probably no need to do it.

**Comment by HoustonPutman** [comment]
@janhoy I decided to go ahead and move to the system of just using the permissions from the TGZ, and removing the overwriting in the dockerfile. This works well, and I just had to fix the permissions for the docker scripts.

**Review by janhoy — APPROVED** [review]

**Pull request merged**